server/tests-py/validate.py

```diff
@@ -274,18 +278,32 @@ def validate_http_anyq(hge_ctx, url, query, headers, exp_code, exp_response):
     else:
         return resp, True
 
+def validate_http_anyq_with_allowed_responses(hge_ctx, url, query, headers, exp_code, allowed_responses):
```

Comment:
It's not immediately obvious to me (though it's probably the case) that the assertion that you defer when doing `assert_graphql_resp_expected( ... skip_assertion=True)` is ever actually checked. 

Would you mind writing a test or two in `tests-py/test_tests.py` which demonstrates:
- `validate_http_anyq_with_allowed_responses` or its parent function actually raises a test failure when none of the nonempty `allowed_responses`
- the same function does what its meant to do (maybe just raising an error) when `allowed_responses` is empty

This will help us from accidentally breaking the test suite and missing test failures

Reply:
Yup, this is a valid concern. I will add these tests and make sure it's well rounded!